scripts/BHM-example.jl

```diff
@@ -1,121 +1,154 @@
-            dτ,
-            targetwalkers,
-            post_step,
+# Finally, we can start the FCIQMC run.
+df, state = lomc!(
+    H, initial_vector;
+    laststep,
+    dτ,
+    targetwalkers,
+    post_step,
 );
 
-# `df` is a `DataFrame` containing the time series data.
+# Here, `df` is a `DataFrame` containing the time series data, while `state` contains the
+# internal state of FCIQMC, which can be used to continue computations.
 
 # ## Analysing the results
 
-# We can plot the norm of the coefficient vector as a function of the number of steps:
-hline([targetwalkers], label="targetwalkers", color=:red, linestyle=:dash)
-plot!(df.steps, df.norm, label="norm", ylabel="norm", xlabel="steps")
-# After some equilibriation steps, the norm fluctuates around the target number of walkers.
+# We can plot the norm of the coefficient vector as a function of the number of steps.
+hline(
+    [targetwalkers];
+    label="targetwalkers", xlabel="steps", ylabel="norm",
+    color=2, linestyle=:dash,
+)
+plot!(df.steps, df.norm, label="norm", color=1)
 
-# Now let's look at estimating the energy from the shift.
-# The mean of the shift is a useful estimator of the shift. Calculating the error bars
-# is a bit more involved as correlations have to be removed from the time series.
-# The following code does that with blocking transformations:
+# After an initial equilibriation period, the norm fluctuates around the target number of
+# walkers.
+
+# Now, let's look at using the shift to estimate the ground state energy of `H`. The mean of
+# the shift is a useful estimator of the energy. Calculating the error bars is a bit more
+# involved as autocorrelations have to be removed from the time series. This can be done
+# with the function [`shift_estimator`](@ref), which performs a blocking analysis.
 se = shift_estimator(df; skip=steps_equilibrate)
 
-# For the projected energy, it a bit more complicated as it's a ratio of fluctuationg
-# quantities:
+# Here, `se` contains the calculated mean and standard errors of the shift, as well as some
+# additional information related to the blocking analysis.
+
+# Computing the error of the projected energy is a bit more complicated, as it's a ratio of
+# fluctuating variables. Thankfully, the complications are handled by the following
+# function.
 pe = projected_energy(df; skip=steps_equilibrate)
 
-# The result is a ratio distribution. Let's get its median and lower and upper error bars
-# for a 95% confidence interval
+# The result is a ratio distribution. We extract its median and the edges of the 95%
+# confidence interval.
 v = val_and_errs(pe; p=0.95)
 
-# Let's visualise these estimators together with the time series of the shift
-plot(df.steps, df.shift, ylabel="energy", xlabel="steps", label="shift")
+# Let's visualise these estimators together with the time series of the shift.
+plot(df.steps, df.shift, ylabel="energy", xlabel="steps", label="energy")
```

Comment:
```suggestion
plot(df.steps, df.shift, ylabel="energy", xlabel="steps", label="shift", margin = 2Plots.cm)
```
I'd prefer to leave "shift" for the legend, as this is the name in the dataframe. It is correct to call the `ylabel` "energy", because all values plotted are energies.

The `margin` argument is the workaround suggested in the linked issue.